README.md

```diff
@@ -1 +1,180 @@
+áp dụng cho ngôn ngữ Java. Đây không phải là một hướng dẫn về cách viết code Java mà là hướng dẫn về cách viết các đoạn code dễ đọc hiểu, tái sử dụng và tái cấu trúc được trong Java.
+
+Không phải mọi nguyên tắc ở đây phải được tuân thủ một cách nghiêm ngặt,
+và thậm chí chỉ có một ít trong số đó được sử dụng phổ biến. Ở đây, nó chỉ là một
+hướng dẫn - không hơn không kém, nhưng chúng được hệ thống hóa thông qua kinh
+nghiệm thu thập được qua nhiều năm của các tác giả của cuốn sách *Clean Code*
+
+Ngành kỹ thuật phần mềm chỉ phát triển được hơn 50 năm, và chúng ta vẫn
+đang học rất nhiều. Một khi kiến trúc phần mềm trở thành phổ biến, có lẽ sau đó
+chúng ta sẽ có thêm nhiều luật lệ khó hơn phải tuân theo. Còn giờ đây,
+hãy để những hướng dẫn này như là một tiêu chuẩn để đánh giá chất lượng các đoạn
+code Java mà bạn và team của bạn tạo ra.
+
+Biết những hướng dẫn này thôi sẽ không thể ngay lập tức làm bạn trở thành một
+lập trình viên phần mềm tốt hơn được, và làm việc với chúng trong nhiều năm
+cũng không có nghĩa bạn sẽ không gặp bất cứ sai lầm nào. Mỗi đoạn code bắt đầu
+như một bản thảo đầu tiên, giống như đất sét được nặn nhào và cho tới cuối cùng
+thì nó sẽ lộ diện hình hài. Cuối cùng, chúng ta gọt tỉa những khuyết điểm khi
+chúng ta xem xét lại nó cùng với các đồng nghiệp.
+Đừng để bản thân bạn bị đánh bại bởi những bản thảo đầu tiên,
+thứ mà vẫn cần phải được chỉnh sửa. Thay vào đó hãy đánh bại những dòng code.
+
+## Biến
+### Sử dụng tên biến có nghĩa và dễ phát âm
+
+**Không tốt:**
+```kotlin
+val yyyymmdstr = SimpleDateFormat("YYYY/MM/DD").format(Date())
+```
+
+**Tốt:**
+```kotlin
+val currentDate = SimpleDateFormat("YYYY/MM/DD").format(Date())
+```
+
+**[⬆ về đầu trang](#mục-lục)**
+
+### Sử dụng cùng từ vựng cho cùng loại biến
+
+**Không tốt:**
+
+```kotlin
+getUserInfo()
+getClientData()
+getCustomerRecord()
+```
+
+**Tốt:**
+```kotlin
+getUser()
+```
+
+**[⬆ về đầu trang](#mục-lục)**
+
+### Sử dụng các tên có thể tìm kiếm được
+Chúng ta sẽ đọc code nhiều hơn là viết chúng. Điều quan trọng là code chúng ta
+viết có thể đọc được và tìm kiếm được. Việc đặt tên các biến *không* có ngữ
+nghĩa so với chương trình, chúng ta có thể sẽ làm người đọc code bị tổn thương
+tinh thần.
+Hãy làm cho các tên biến của bạn có thể tìm kiếm được.
+
+**Không tốt:**
+```kotlin
+// 86400000 là cái quái gì thế?
+setTimeout(blastOff, 86400000)
+
+```
+
+**Tốt:**
+```kotlin
+// Khai báo chúng như một biến hằng global.
+const val MILLISECONDS_IN_A_DAY = 86400000
+
+setTimeout(blastOff, MILLISECONDS_IN_A_DAY)
+
+```
+
+**[⬆ về đầu trang](#mục-lục)**
+
+### Sử dụng những biến có thể giải thích được
+**Không tốt:**
+```kotlin
+val pattern1 = Regex("g([ee]+)ks?")
+checkPatternMatched(pattern1.matches("geeeeks"), pattern1.matches("geeksforgeeks"))
+```
+
+**Tốt:**
+```kotlin
+val geeksPattern = Regex("g([ee]+)ks?")
+checkPatternMatched(pattern1.matches("geeeeks"), pattern1.matches("geeksforgeeks"))
+```
```

Comment:
Reference replace by [this example](https://github.com/leonardolemie/clean-code-java#use-explanatory-variables), It looks more than clearly